csharp/Ice/async/Server.cs

```diff
@@ -3,60 +3,55 @@
 //
 
 using System;
+using System.Threading;
+using System.Threading.Tasks;
 
 public class Server
 {
-    public static int Main(string[] args)
+    public static async Task<int> Main(string[] args)
```

Comment:
Visual Studio 2015 doesn't allow async main

Reply:
Fixed and tested with VS 2015.